### PR TITLE
npm:element-resize-detector override

### PR DESCRIPTION
**Pull request by MeirionHughes**

### CODE REVIEW

File: package-overrides/npm/element-resize-detector@1.1.0.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "main": "dist/element-resize-detector.min",
```
Comment (guybedford) [inline package-overrides/npm/element-resize-detector@1.1.0.json]: It's best to use the unminified source on jspm for debugging purposes.

**Comment by MeirionHughes** [comment]
Okay, updated it.

File: package-overrides/npm/element-resize-detector@1.1.0.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "main": "dist/element-resize-detector",
+  "format": "cjs",
+  "shim": {
+    "dist/element-resize-detector": {
+      "exports": "utils"
+    }
+  }
```
Comment (guybedford) [inline package-overrides/npm/element-resize-detector@1.1.0.json]: Is this shim really necessary here? Does it not work ok without it?

**Comment by guybedford** [comment]
Thanks, just noticed it should in theory work as CommonJS through the dist file though?

File: package-overrides/npm/element-resize-detector@1.1.0.json (hunk 2 above)
Comment (MeirionHughes) [inline package-overrides/npm/element-resize-detector@1.1.0.json]: yeah, during bundling it says it cannot resolve that utils, all the other exports are fine though.

**Comment by MeirionHughes** [comment]
yeah I think it can work okay without that too. one mo

**Comment by MeirionHughes** [comment]
Anything else needed?

**Comment by guybedford** [comment]
Yes I wanted to test it. I tried it out and this works and bundles fine without the shim in jspm 0.17. I take it the reason you needed it is because you are running jspm 0.16 here?

**Comment by MeirionHughes** [comment]
ahh perhaps. let me update my global jspm.

**Comment by guybedford** [comment]
The upgrade path is breaking, so I certainly wouldn't force that on you. Happy to merge this as-is, let me know.

**Comment by MeirionHughes** [comment]
yeah, if you don't mind. I think most people will be on 0.16 if they do `npm install jspm -g`

**Comment by guybedford** [comment]
One other thing in this override - `batch-processor` is still a dependency, but doesn't need to be using the dist version here. This can be handled by amending this with `"dependencies": {}`.

**Comment by MeirionHughes** [comment]
unfortunately, it would appear the shim doesn't work properly; I had been using it with the bundler (which worked). When used without any bundling (aurlia skeleton app). It complains: 

> uncaught exception: ReferenceError: utils is not defined
> 	Evaluating http://localhost:9000/jspm_packages/npm/element-resize-detector@1.1.0/dist/element-resize-detector.js

**Comment by simonfox** [comment]
We are seeing the same issue since this was merged. Can we get this rolled back? We too were using this in Aurelia without issue prior to the override being added.

**Comment by MeirionHughes** [comment]
Requested rollback. 
I'd be interested to know how you got it to work without the shim; it complains it can't find "utils" for me. :/

**Comment by simonfox** [comment]
@MeirionHughes we didn't need to do anything special....just `jspm install npm:element-resize-detector` and then usage (abridged)...

```js
import erd from 'element-resize-detector';

export class ThatUsesResizeDetection {
  constructor(element, eventAggregator, singleEventQueue) {
    this.elementResizeDetector = erd({ strategy: 'scroll' });
  }
}
```

**Comment by MeirionHughes** [comment]
hmm is that jspm 0.17?

**Comment by simonfox** [comment]
@MeirionHughes nope, `0.16.30`

**Comment by MeirionHughes** [comment]
how odd; all works fine on my home machine. my work machine was complaining about the "utils" during bundling. I'll have to update everything and try again. Sorry the grief. :P

**Comment by simonfox** [comment]
@MeirionHughes no worries :+1:

**Comment by MeirionHughes** [comment]
you'd be happy with the package alias though right? 
https://github.com/jspm/registry/pull/819/commits/f6f95957ccd017d5b146213d36506d86733cd38c

**Comment by simonfox** [comment]
@MeirionHughes yes definitely

**Pull request merged**